server/service/transport.go

```diff
@@ -294,6 +294,30 @@ func hostListOptionsFromRequest(r *http.Request) (fleet.HostListOptions, error)
 		hopt.SoftwareTitleIDFilter = &sid
 	}
 
+	softwareStatus := fleet.SoftwareInstallerStatus(strings.ToLower(r.URL.Query().Get("software_status")))
+	if softwareStatus != "" {
+		if !softwareStatus.IsValid() {
+			return hopt, ctxerr.Wrap(
+				r.Context(), badRequest(fmt.Sprintf("Invalid software_status: %s", softwareStatus)),
+			)
+		}
+		if hopt.SoftwareTitleIDFilter == nil {
+			return hopt, ctxerr.Wrap(
+				r.Context(), badRequest(
+					"Missing software_title_id (it must be present when software_status is specified)",
+				),
+			)
+		}
+		if hopt.TeamFilter == nil {
+			return hopt, ctxerr.Wrap(
+				r.Context(), badRequest(
+					"Missing team_id (it must be present when software_status is specified)",
```

Comment:
Does that mean that you can't filter software status on no team? Or the caller needs to explicitly set the team id to 0 for that?

Reply:
we learned yesterday that software is not allowed on "no team", teams-only feature 😱

Reply:
Huh, that feels... weird? Seems inconsistent with other features to me. I assume we have good reasons for that, though.